vitest.config.ts

```diff
@@ -18,6 +25,12 @@ export default defineConfig({
       'extensions/**/*.spec.{ts,tsx}',
       'extensions/**/*.test.{ts,tsx}',
     ],
+    environmentOptions: {
+      jsdom: {
+        resources: 'usable',
```

Comment:
this was the key change to tell jsdom to load images
and onload handlers for image element work fine now